Pull Request for Issue 2505: Build AMHDF5Error

mButlerDM opened the pull request:
Implementation of AMHDF5Error for #2505.

Commit:
Added:

Added in the AMHDF5Error class.

Commit:
Added:

Non function implementation of a C macro. Currently getting data into
the proper memory space is problematic in a C++ environment.

Added printErrorStack and clearErrorStack as basic functions.

Commit:
Merge branch 'Issue2486' into Issue2505

Conflicts:
	acquamanCommon.pri
	source/ui/PGM/PGMPersistentView.cpp

Commit:
Modified:

Cleaned up and removed commented code that is not required.

Left in the HDF5 macro function with explanation on its purpose.

mButlerDM commented:
@dretrex I imagine this can be added `AMHDF5File` but its usefulness right now is limited.

dretrex commented on source/dataman/HDF5/AMHDF5Error.cpp:
```diff
@@ -0,0 +1,68 @@
+		return false;
+	}
+
+	herr_t printStatus = H5Eprint2(errorStack, stdout);
```
Is it all possible to be able to provide some kind of `FILE *` that would allow you to get this information into a QString to be added to an error monitor at all?

mButlerDM replied:
I think this is possible. A memstream would avoid having to have a file handle but it would take a bit of figuring. I'll keep looking at this option.

dretrex commented on source/dataman/HDF5/AMHDF5Error.cpp:
```diff
@@ -0,0 +1,68 @@
+//	H5Eget_major(err_desc->maj_num);
+//	H5Eget_minor(err_desc->min_num);
+
+//	memcpy((char *)client_data, majorError, 32 * sizeof(char));
```
This might be worth playing with.  If you got the client data copied into a `char *` buffer you should be able to get that into a string and return it.

mButlerDM replied:
I tried this for my first attempt but had trouble getting memcpy to move data over. I'm guessing it was memory protection for C++ classes. I'll look at this option a bit more closely.

Commit:
Modified:

AMHDF5 output to stderr is turned off.

AMHDF5 output is now redirected to a stream and captured for output.

Now will work on cleaning up unneeded code and setting up output to
AMErrorMon.

Commit:
Modified:

All AMHDF5 classes to use error calls.

Cleaned up AMHDF5Error to use AMErrorMonitor and do additional error
checking internally.

mButlerDM commented:
@dretrex I got error output to `AMErrorMonitor` working with streams. There is likely a better way to reference the `AMHDF5Error` object/instance between the classes.

Still have a few small things to clean up.

Commit:
Modified:

Removed commented code, errors, whitespace, and filled in AMErrorMon
calls within AMHDF5Error.
